contents/handbook/engineering/how-we-review.md

```diff
@@ -0,0 +1,42 @@
+---
+title: How we review PRs
+sidebar: Handbook
+showTitle: true
+---
+
+Almost all PRs made to repositories on PostHog will need a review from another engineer. We do this because almost every time we review a PR we find a bug, a performance issue, unnecessary code or UX that could have been confusing.
+
+## How to review
+
+1. Have a flick through the code.
+  - What to look for:
+    - Code that does the wrong thing or will produce bugs
+    - Code that you think will cause performance issues
+    - Are there tests for all of the new functionality, and do they test the right thing?
+    - Any security issues or project leakage?
+  - What _not_ to look for:
+    - Formatting issues (prettier should handle this, raise a PR to fix that)
+    - "I would have done it differently" (Unless the code is completely incomprehensible or unreadable, or will cause us massive harm long term, as long as it works it's good enough.)
```

Comment:
Not-a-comment: Curious to hear others thoughts on this - I've used PRs in the past as a learning-teaching tool, so approaching it like this is new for me. :)

Reply:
I think it's a balance. At our stage I think this (the 'ship it' mentality) makes sense as long as the compromises are understood and the timeline to fix is also agreed upon. If it's a: this gets the feature out the door now and we'll clean it up within 2 weeks, then I am all for the ship it mentality. You might see in the 2 weeks that the feature is not worth the investment.

Later when we have a bunch of junior/learning engineers working on things I think this will change and morph more into the learning and growing opportunities like you mentioned @macobo. Just because we'll have so much more bandwidth for that and it will improve our overall shipping capabilities.

Reply:
In general, I agree with @fuziontech that this should be a balance. Just to offer my own POV, I've found very helpful to get these sort of suggestions in areas outside my expertise (e.g. @mariusandra suggesting refactors from `useEffect` / `useState` to kea). Perhaps to add the suggestions/teachings should be fine, but they shouldn't be a blocker to an approving review?